src/uu/tac/src/tac.rs

```diff
@@ -80,12 +80,49 @@ pub fn uu_app() -> App<'static, 'static> {
         .arg(Arg::with_name(options::FILE).hidden(true).multiple(true))
 }
 
-fn tac(filenames: Vec<String>, before: bool, _: bool, separator: &str) -> i32 {
-    let mut exit_code = 0;
+fn buffer_tac(data: &[u8], before: bool, separator: &str) -> std::io::Result<()> {
     let mut out = stdout();
+
+    // Convert the line separator to a byte sequence.
     let sbytes = separator.as_bytes();
     let slen = sbytes.len();
 
+    // If there are more characters in the separator than in the data,
+    // we can't possibly split the data on the separator. Write the
+    // entire buffer to stdout.
+    let dlen = data.len();
+    if dlen < slen {
+        return out.write_all(data);
+    }
+
+    // Iterate over each byte in the buffer in reverse. When we find a
+    // line separator, write the line to stdout.
+    //
+    // The `before` flag controls whether the line separator appears at
+    // the end of the line (as in "abc\ndef\n") or at the beginning of
+    // the line (as in "/abc/def").
+    let mut following_line_start = data.len();
+    for i in (0..dlen - slen + 1).rev() {
+        if &data[i..i + slen] == sbytes {
```

Comment:
This is not correct when the separator is multiple characters and it overlaps: e.g.`printf "a///" | tac -s "//"`. In `///` it detects a separator twice, once from the second to the third character and once from the first to the second. One way to fix this would be to rewrite this as a loop and subtract `slen` from `i` every time it finds a match. However, I think it would be easier to add a dependency to `memchr`, which provides a (probably faster) implementation: https://docs.rs/memchr/2.4.1/memchr/memmem/fn.rfind_iter.html

Reply:
It seems that the test case you provide is already broken on the `master` branch.

GNU `tac`:
```
$ printf "a///" | tac -s "//"
a///
```
`uutils` `master` branch:
```
$ printf "a///" | ./target/debug/coreutils tac -s "//"
/a//
```
`uutils` this branch:
```
$ printf "a///" | ./target/debug/coreutils tac -s "//"
/a//
```

Would it be acceptable to open a separate pull request to resolve that issue? Or do you prefer updating this pull request to fix that as well?

Reply:
No, in this case a separate pull request is fine